### PR TITLE
Product recommendation

### PR DESCRIPTION
**Pull request by linuskar**

### CODE REVIEW

File: src/main/java/org/team6/controller/RecommendationsPageController.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+    private void initializeToggleButtons() {
+        recommendationsToggleButton.setSelected(RecommendationsBackend.isPersonalRecommendationsOn());
+        personalToggleButton.setDisable(!RecommendationsBackend.isPersonalRecommendationsOn());
+    }
+
+    private void initializeGeneralRecommendations(int products, int tips) {
+        try {
+            FXMLLoader loader = new FXMLLoader(getClass().getResource("/org/team6/view/RecommendationContent.fxml"));
+            generalRecommendationsFlowPane = loader.load();
```
Comment (Caramajau) [inline src/main/java/org/team6/controller/RecommendationsPageController.java:126]: The initialize methods could maybe have even more functional decomposition.

File: src/main/java/org/team6/controller/SettingsPageController.java [hunk 2]
```diff
@@ -19,7 +20,10 @@
 import java.util.Map;
 import java.util.ResourceBundle;
 
-public class SettingsPageController implements Initializable {
+import org.team6.model.Observer;
+import org.team6.model.RecommendationsBackend;
+
+public class SettingsPageController implements Initializable, Observer {
```
Comment (Caramajau) [inline src/main/java/org/team6/controller/SettingsPageController.java:26]: Observer should maybe be called ProductRecommendationObserver to make it more clear that it is different

File: src/main/java/org/team6/model/RecommendationsBackend.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                List<Product> relevantProducts = products.stream()
+                        .filter(product -> product.getCategory().equals(usage.getCategory()) && product.isEnergyEfficient())
+                        .collect(Collectors.toList());
+
+                // Compare with current products
+                for (Product currentProduct : currentProducts) {
+                    if (currentProduct.getCategory().equals(usage.getCategory())) {
+                        relevantProducts = relevantProducts.stream()
+                                .filter(product -> product.getEnergySavings(currentProduct) > 0)
+                                .collect(Collectors.toList());
```
Comment (Caramajau) [inline src/main/java/org/team6/model/RecommendationsBackend.java:65]: Could perhaps be more functional decomposed.

File: src/main/java/org/team6/view/PageStarter.java [hunk 4]
```diff
@@ -15,7 +16,13 @@
 
 public class PageStarter {
     private static AnchorPane homePane;
+    @FXML
+    private static AnchorPane energyInsightsPane;
+    @FXML
     private static AnchorPane settingsPane;
+    @FXML
+    private static AnchorPane recommendationsPane;
```
Comment (Caramajau) [inline src/main/java/org/team6/view/PageStarter.java:24]: FXML tags should be unnecessary in PageStarter

**Review by Caramajau — APPROVED** [review]

File: src/main/java/org/team6/controller/RecommendationsPageController.java (hunk 1 above)
Comment (linuskar) [reply src/main/java/org/team6/controller/RecommendationsPageController.java:126]: will look into it

File: src/main/java/org/team6/controller/SettingsPageController.java (hunk 2 above)
Comment (linuskar) [reply src/main/java/org/team6/controller/SettingsPageController.java:26]: will look into it

File: src/main/java/org/team6/model/RecommendationsBackend.java (hunk 3 above)
Comment (linuskar) [reply src/main/java/org/team6/model/RecommendationsBackend.java:65]: will look into it

**Review by Figer-op — COMMENTED** [review]
| approved |

**Review by Figer-op — APPROVED** [review]
actually aproved now

**Pull request merged**